### ISSUE TITLE
Installation Service.

### ISSUE THREAD

**Issue by alanmarsham** [kiswa/TaskBoard#388]
Hi,

I am a newbie here and would like to know if you provide and installation service.
If so, how much do you charge?
Do you take PayPal ?

Thank you.

**Comment by kiswa** [comment]
Currently there is not such a thing.

I have considered creating a hosted version of TaskBoard so people can just sign up and get going, but I'm not sure what the interest would be for that.

**Comment by amunizp** [comment]
Yes, all the other competitors that I found that provide this are not free (as in freedom) software. A way to try before you install in server would be great.

**Comment by alanmarsham** [comment]
Thanks for your response and the fab software.

May I make a suggestion which I realize may not be feasible for you.

Maybe a version you can download that can be loaded straight into a VM.  (Easy solution for small and medium sized businesses, who have no expert knowledge of setting up software).
I use Microsoft VMs on MS Server 2012, I use Linux VMs on VMWare Workstation.
You could maybe sell a pre-set up version that is a single file that loads straight into a VM host.    
 <<<  Just a thought  >>>. 
I personally would pay money for that.  It takes my set up cost down.  This is also not being done by anyone else.  Maybe there is a reason for this which i am not aware of.

Thanks for listening.

Alan.

**Comment by kiswa** [comment]
@amunzip There is a demo at https://taskboard.matthewross.me/demo so people can try it out before using their own setup. Maybe I don't make that obvious enough.

@alanmarsham That's an interesting idea I hadn't considered before. I'll have to look into what that would entail.

My original idea was just to have a dedicated site for people to sign up and pay a small fee for a hosted instance that they would then have complete control over.

**Comment by rautamiekka** [comment]
Docker is 1 of the typical methods in use, the only one I can recall right away.

**Comment by hbayindir** [comment]
@kiswa Actually I use taskboard on my personal office computer to keep data on my computer, so hosting taskboard will not make it better for me. Also,providing an hosted service will force you to compete with Trello and the like IMHO, do you want that?

Providing VM (possibly Vagrant) and Docker images will be a very nice alternative to more novice users to run and possibly update taskboard with a signle command will be much more interesting for small / medium corparations or companies which needs strict data locality rules.

**Comment by triode3** [comment]
I know this is old, but the taskboard demo looks exactly like what I wanted to use. However, installation always fails for me, and the application does not work. As an experienced sysadmin speaking, I think you could give us details about how you made 0.3 work on your demo site (e.g. which versions of software are you using on the server), or move forward with the new version. 

As for a live site for a "try before you install", I would be tempted to agree with others and say just provide a working VM/Container. 

Thanks for all of your work.

**Comment by hbayindir** [comment]
@triode3 Which distro are you running? I can guide you if you wish.

**Comment by DavidH48** [comment]
I'm new here too.

The install instructions are not clear (enough) to me.

1.  Do I download a zip file
2.  Unzip all files
3.  Upload all files to a subdirectory on my site
4.  Login to what???

Using the above as a starting point, if it's anywhere near correct, can it be expanded to further clarify installing and running?

Thank you,
– David

PS: I really like this product because I can extend it with other project management ideas and graphics.

**Comment by catxeger** [comment]
It's not clear, but if you download the zipfile and expand it, there's a file 'README.md' which has more details.

The general gist is:
1) Meet all pre-reqs (webserver, php, sqlite...)
2) Put zipfile in subdirectory of site, extract
3) Run `./build/composer.phar install` in that same subdirectory (optionally do other bits for production)
4) Connect to the site 'normally' (whatever the path to that subdir would be), and log in using admin/admin [and change that right after].

There's a bunch of other bits, but that should, at least, get you going.

**Comment by hbayindir** [comment]
After installing everything up, one shall set that the `/api` subfolder as writable and set `AllowOverrides` to `all` or `true` to make sure the `/api` folder is writable and set-up as intended by the developer. I don't remember the exact settings, and I cannot access to the server right now.

**Comment by palesaj** [comment]
I tried everything, but it won't work. The composer.phar comes with a message:
Warning: This development build of composer is over 60 days old. It is recommended to update it by running "C:\xampp\htdocs\taskboard\build\composer.phar self-update" to get the latest version.
Self-update does not work.

The demo look very nice and I am eager to get it work. So please, be more specific what to do, or create an out of the box app.

**Comment by dmagaha** [comment]
I cloned the git repo and followed the instructions in README.md.  I get the app to run in apache2 web server with php7.2 but no access to the DB seems to be happening.  I was able to get it to run locally on my MAC but when I tried on an ubuntu vm I get none of the dynamics.  The api directory is writable because I got a different error locally and made sure it was running.  Any help or suggestions are appreciated.
